src/software/ai/navigator/path_manager/obstacle_path_manager.cpp

```diff
@@ -0,0 +1,79 @@
+#include "software/ai/navigator/path_manager/obstacle_path_manager.h"
+
+/**
+ * This file contains the implementation of the obstacle path manager
+ * which uses obstacles to arbitrate between paths
+ * ret a path consisting of only the start and destination
+ * points.
+ */
+
+
+ObstaclePathManager::ObstaclePathManager(std::unique_ptr<PathPlanner> path_planner)
+    : path_planner(std::move(path_planner))
+{
+}
+
+const std::map<int, Path> ObstaclePathManager::getManagedPaths(
+    const std::map<int, PathObjective> &objectives, const Rectangle &navigable_area,
+    const std::vector<Obstacle> &static_obstacles)
+{
+    std::map<int, Path> retval;
+
+    std::vector<Obstacle>
+        current_velocity_obstacles;  // velocity obstacles used to avoid collisions
+
+    // cache dynamic params
+    double robot_obstacle_inflation = Util::DynamicParameters->getNavigatorConfig()
+                                          ->RobotObstacleInflationFactor()
+                                          ->value();
+    double velocity_obstacle_inflation = Util::DynamicParameters->getNavigatorConfig()
+                                             ->VelocityObstacleInflationFactor()
+                                             ->value();
+
+    for (auto const &current_objective : objectives)
+    {
+        // find path with relevant obstacles
+        std::vector<Obstacle> path_obstacles = getObstaclesFromOtherObjectives(
+            objectives, current_objective.first, robot_obstacle_inflation);
+        path_obstacles.insert(path_obstacles.begin(), current_velocity_obstacles.begin(),
```

Comment:
unless we have a specific reason for it, I think that inserting at the end of the list is more efficient because then elements don't need to be shifted. Same comment goes for the case directly below.

Reply:
done